e2e/src/usecases/WalletConnectV2.js

```diff
@@ -93,283 +109,312 @@ export default WalletConnect = () => {
-        },
-      },
-    }
-    const params = [walletAddress, JSON.stringify(typedData)]
-    const [session] = walletConnectClient.session.map.values()
-    walletConnectClient
-      .request({
-        topic: session.topic,
-        chainId: 'eip155:44787',
-        request: {
-          method: 'eth_signTypedData',
-          params,
-        },
+      it('Then is able to send a transaction (eth_sendTransaction)', async () => {
+        let txHash
+        const tx = await formatTestTransaction(walletAddress, web3Library)
+        const [session] = walletConnectClient.session.map.values()
+        walletConnectClient
+          .request({
+            topic: session.topic,
+            chainId: 'eip155:44787',
+            request: {
+              method: 'eth_sendTransaction',
+              params: [tx],
+            },
+          })
+          .then((result) => {
+            txHash = result
+          })
+
+        await waitFor(element(by.text(`${dappName} would like to send a Celo transaction.`)))
+          .toBeVisible()
+          .withTimeout(15 * 1000)
+        await verifySuccessfulTransaction('Send transaction', tx)
+
+        // Wait for transaction and get receipt
+        const { status, from, to } = await kit.connection.getTransactionReceipt(txHash)
+
+        jestExpect(status).toStrictEqual(true)
+        jestExpect(from).toStrictEqual(walletAddress)
+        jestExpect(to).toStrictEqual(walletAddress)
       })
-      .then((result) => {
-        signature = result
+
+      it('Then is able to sign a transaction (eth_signTransaction)', async () => {
+        let signedTx
+        const tx = await formatTestTransaction(walletAddress, web3Library)
+        const [session] = walletConnectClient.session.map.values()
+        walletConnectClient
+          .request({
+            topic: session.topic,
+            chainId: 'eip155:44787',
+            request: {
+              method: 'eth_signTransaction',
+              params: [tx],
+            },
+          })
+          .then((result) => {
+            signedTx = result
+          })
+
+        await waitFor(element(by.text(`${dappName} would like to sign a Celo transaction.`)))
+          .toBeVisible()
+          .withTimeout(15 * 1000)
+        await verifySuccessfulTransaction('Sign transaction', tx)
+
+        // TODO: keep only Viem branch after feeCurrecny estimation is ready
+        if (web3Library === 'viem') {
+          // TODO: assert transaction signer address if once Viem could provide it
+          const recoveredTx = parseTransactionCelo(signedTx)
+          jestExpect(recoveredTx.nonce).toEqual(hexToNumber(tx.nonce))
+          jestExpect(recoveredTx.to).toEqual(tx.to)
+          jestExpect(recoveredTx.gas).toEqual(BigInt(tx.gas))
+          jestExpect(recoveredTx.value).toEqual(BigInt(tx.value))
+        } else {
+          const [recoveredTx, recoveredSigner] = recoverTransaction(signedTx)
```

Comment:
`recoverTransaction` only works with legacy transactions? Or would it help to update `@celo/...` libs?

Reply:
Yes, if we update Celo libs, it possibly should work with both legacy and CIP-42.

But the update is not looking straightforward, I think it's better to make a separate PR for that.

Reply:
oh yes, a separate PR is definitely warranted :D